src/ase/cw/view/ServerFrame.java

```diff
@@ -16,38 +17,46 @@
 public class ServerFrame extends JFrame implements ActionListener, ServerFrameView {
 
     private JTextArea textArea = new JTextArea("");
+    private JTextArea statusArea = new JTextArea("");
     private JButton breakButton = new JButton("On-Break");
     private JButton restartButton = new JButton("Restart");
+    private Pausable pausable;
     private int serverId;
-    private OrderController oc;
 
+    private int dimHeight=150;
+    private int dimWidth=300;
+    private int gapX = 10;
+    private int gapY = 40;
 
     // Frame constructor
-    public ServerFrame(int id, JFrame parentFrame, OrderController controller) {
+    public ServerFrame(int id, JFrame parentFrame, Pausable pausable) {
         this.serverId = id;
         this.textArea.setEditable(false);
-        this.oc = controller;
+        this.statusArea.setEditable(false);
+        this.pausable = pausable;
         this.setIconImage(Toolkit.getDefaultToolkit().getImage(getClass().getClassLoader().getResource("coffee.png")));
         this.setTitle("Server "+serverId);
         this.setName("QueueFrame "+this.getTitle());
-        this.setPreferredSize(new Dimension(300, 200));
-        this.setLocation(parentFrame.getX()+parentFrame.getWidth()+(id*10), parentFrame.getY()+(id*10));
+        this.setPreferredSize(new Dimension(dimWidth, dimHeight));
+        this.setLocation(parentFrame.getX()+parentFrame.getWidth()+gapX, parentFrame.getY()+(serverId-1)*(gapY+dimHeight));
```

Comment:
please update this from my last pull request as this puts frames with high id numbers outside of viewing area or stacks them up

Reply:
okay fixed.